src/commands/cmd_txn.cc

```diff
@@ -45,14 +45,14 @@ class CommandMulti : public Commander {
 class CommandDiscard : public Commander {
  public:
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
-    auto reset_watch = MakeScopeExit([svr, conn] { svr->ResetWatchedKeys(conn); });
-
     if (!conn->IsFlagEnabled(Connection::kMultiExec)) {
       *output = redis::Error("ERR DISCARD without MULTI");
       return Status::OK();
     }
 
+    auto reset_watch = MakeScopeExit([svr, conn] { svr->ResetWatchedKeys(conn); });
     conn->ResetMultiExec();
```

Comment:
Not sure if i should change it like this (consistent with EXEC)
```suggestion
    auto reset_multiexec = MakeScopeExit([conn] { conn->ResetMultiExec(); });
```

Reply:
Just keeping the original form is fine here.